src/controller/CHIPDeviceController.cpp

```diff
@@ -364,16 +364,15 @@ CHIP_ERROR DeviceController::DisconnectDevice(NodeId nodeId)
 {
     ChipLogProgress(Controller, "Force close session for node 0x%" PRIx64, nodeId);
 
-    OperationalDeviceProxy * proxy = mSystemState->CASESessionMgr()->FindExistingSession(GetPeerScopedId(nodeId));
-    if (proxy == nullptr)
+    if (SessionMgr()->MarkSessionAsDefunctMark(GetPeerScopedId(nodeId), MakeOptional(Transport::SecureSession::Type::kCASE)))
     {
-        ChipLogProgress(Controller, "Attempted to close a session that does not exist.");
         return CHIP_NO_ERROR;
     }
 
-    if (proxy->IsConnected())
+    OperationalSessionSetup * proxy = mSystemState->CASESessionMgr()->FindExistingSessionSetup(GetPeerScopedId(nodeId));
```

Comment:
Follow-up: re-evaluate whether this code still makes sense or whether it should just return success in the "not connected but connecting" case.

Reply:
Oh I don't think it does make any sense, I was trying to keep parity knowning that in https://github.com/project-chip/connectedhomeip/issues/20452 this will get removed